diff --git a/repo/src/main/java/org/apache/commons/lang3/math/NumberUtils.java b/repo/src/main/java/org/apache/commons/lang3/math/NumberUtils.java
index 2522aca..c9f2b4d 100644
--- a/repo/src/main/java/org/apache/commons/lang3/math/NumberUtils.java
+++ b/repo/src/main/java/org/apache/commons/lang3/math/NumberUtils.java
@@ -1729,9 +1729,6 @@ public static boolean isParsable(final String str) {
         if (StringUtils.isEmpty(str)) {
             return false;
         }
-        if (str.charAt(str.length() - 1) == '.') {
-            return false;
-        }
         if (str.charAt(0) == '-') {
             if (str.length() == 1) {
                 return false;
